amazon-kinesis-client/src/test/java/software/amazon/kinesis/lifecycle/ShardConsumerSubscriberTest.java

```diff
@@ -311,6 +309,80 @@ public void restartAfterRequestTimerExpiresTest() throws Exception {
 
     }
 
+    @Test
+    public void restartAfterRequestTimerExpiresAfterInitialSubscriptionFailsTest() throws Exception {
```

Comment:
How long does this test take to run? Generally we want to avoid timing-related tests as it increases the build time, but in this case that might be the best way to go about it (not close enough to the issue to say)

Reply:
I simplified the test. It takes around 60 to 80 ms to run in general which is in line with other tests in the class.